service/resource/endpoint/create.go

```diff
@@ -91,3 +62,32 @@ func (r *Resource) ProcessCreateState(ctx context.Context, obj, createState inte
 
 	return nil
 }
+
+func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
```

Comment:
@kopiczko do you think the private `new*Change` methods should return a specific type or should they also simply return `interface{}`. I Migrated so far by keeing `interface{}`. We just saw that you in azure-operator used `[]Deployment` as specific type. How should we do this and why?

Reply:
I prefer using a proper type whenever we can.  You can switch that from `[]Deployment` to `interface{}` and this still compiles and works the same. Just a little bit more type safety.

I find this especially beneficial when `New*Patch` is dump. It protects against returning nil when the `new*Change` is expected to return a non-{pointer,struct,map} value.

But this is interchangeable and no real benefit besides type safety.

> I Migrated so far by keeing interface{}.

I see no issue with that. I think I did the same in cert-oeprator. Let's take a common approach in new Resources though. I vote for proper type.

Reply:
> I vote for proper type.

Agreed.